### PR TITLE
added rest-api documentation

### PR DESCRIPTION
**Pull request by rafaelmartins**
Changes proposed in this pull request:

- added `ovirt-engine-api-model`'s gh-pages branch as a subrepo, to deploy rest-api docs to the website. This is not the final solution, it is just a PoC to show how it works, I'm not sure if `/documentation/rest-api/` is the best place for the docs, and if we should mention `master` docs on the website.

I confirm that this pull request was submitted according to the [contribution guidelines](/CONTRIBUTING.md): @rafaelmartins

This pull request needs review by: @sandrobonazzola @jhernand @phoracek @MarSik @mykaul (anyone else is free to review)

### CODE REVIEW

File: source/documentation/index.html.haml [hunk 1]
```diff
@@ -47,6 +47,10 @@ wiki_last_updated: 2014-04-26
       - [Quick Start Guide](/documentation/quickstart/quickstart-guide/)
       - [oVirt Administration Guide](/documentation/admin-guide/administration-guide/)
       - [oVirt User Guide](/documentation/user-guide/user-guide/)
+      - REST API:
+          [oVirt 4.1](/documentation/rest-api/4.1/),
+          [oVirt 4.0](/documentation/rest-api/4.0/),
+          [oVirt master (development version, not recomended for end users)](/documentation/rest-api/master/)
```
Comment (rollandf) [inline source/documentation/index.html.haml]: recomended->recommended

Comment (rafaelmartins) [inline source/documentation/index.html.haml]: fixed.

**Comment by bproffitt** [comment]
This is breaking in CI because files were checked out of the source/ directory, which fundamentally breaks things. There is a .git/ directory and a lot more vital files under this directory. Was this running on your local machine before you committed it @rafaelmartins ?

**Comment by garrett** [comment]
1. Was this tested on someone's machine locally before being committed? (That is, was this actually built successfully? Just running the server isn't always enough, as it doesn't necessarily build all files.)
2. It's a bad idea in general to put non-document files in `source/`* (and in this case, that includes the `.git/` subdirectory) — Middleman does not like this

**Comment by garrett** [comment]
Also:

1. Why is this a submodule?
2. Why is it a dump of (huge, that is > 2MB) HTML pages?

**Comment by rafaelmartins** [comment]
@bproffitt yes, it runs using `./run-server.sh`.

@garrett I do not ran a full build, but looking at travis I know why it breaks. that's because we are lacking an index in the subrepo. I sent a pull request to the subrepo, adding it. the issue will be fixed as soon as it gets merged, and I update the subrepo here.

**Comment by jhernand** [comment]
If it is that problematic to have this directly in the oVirt site, I'd suggest to just have the index page in the oVirt site, and then links to the external site.

**Comment by rafaelmartins** [comment]
I think that having docs hosted in the main project domain should be a priority. I can't see a reasonable argument to do not integrate it this way.

**Comment by garrett** [comment]
There was a previously discussion about API docs belonging on a separate subdomain (afer a discussion about the various pros and cons) @ https://github.com/oVirt/ovirt-site/issues/398. 

Suggested solution @ https://github.com/oVirt/ovirt-site/issues/398#issuecomment-244046989 also has many benefits, such as being in a uniform UI, would be searchable, and can be accessed offline as well.

**Comment by rafaelmartins** [comment]
So, you are "suggesting" to "rewrite" the docs, to use a standalone tool, that needs to be forked, "adapted" and deployed, instead of integrating what we have already, and let users without "official" docs meanwhile?

I have no problem with subdomains, but we need something that is available right now. Users are without decent docs for very long already. We should not step back just because something is not ideal. We can improve it later, but the current situation is not acceptable, IMO.

**Comment by bproffitt** [comment]
If you can get this fixed with the patch you do in the subdomain, I would do that now, and we can look at the #398 method later. I would rather have working documentation up ASAP, too.

**Comment by rafaelmartins** [comment]
Ok, I'll comment here when it is done.

Thanks

**Comment by rafaelmartins** [comment]
sorry, I broke this by mistake, and it was outdated anyway. will reopen a correct pr soon

**Pull request closed without merging**